File: 백준/Bronze/9506. 약수들의 합/약수들의 합.js
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n")
// let input = fs.readFileSync("./EX.txt").toString().trim().split("\n")
const N = input.length

function divisor(x) {
    let answer = x + ' = ' + '1'
    let total  = 1
    
    for (let i = 2; i < x; i++ ){
        if (x % i == 0) {
            total += i
            answer += ' + '
            answer += i
        }
    }
    
    if (total == x) {
        return answer
    } else {
        answer = x + " is NOT perfect."
        return answer
    }
}

for (let i = 0; i < N; i++ ) {
    if (input[i] != -1) {
        answer = divisor(input[i])
        console.log(answer)
    }
}

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.py
import sys
input = sys.stdin.readline

def divisor(x):
    div = []

    for i in range(1, x):
        if x % i == 0:
            div.append(i)

    if sum(div) == x:
        answer = str(x) + ' = '
        cnt = 0
        while cnt < len(div):
            answer += str(div[cnt])
            if cnt != len(div)-1:
                answer = answer + " + "
            cnt += 1
        return answer

    else:
        answer = str(x) + " is NOT perfect."
        return answer

while True:
    x = int(input())
    if x == -1:
        break

    result = divisor(x)
    print(result)